assets/src/components/searchForm.tsx

```diff
@@ -146,25 +160,93 @@ const Filters = ({
   )
 }
 
+export const MapSearchCombobox = ({
+  autocompleteController,
+  autocompleteId,
+  dispatch,
+  formSearchInput,
+  onSubmit,
+  query,
+  setAutocompleteEnabled,
+}: {
+  autocompleteController: MutableRefObject<GroupedAutocompleteControls | null>
+  autocompleteId: string
+  dispatch: Dispatch
+  formSearchInput: React.MutableRefObject<HTMLInputElement | null>
+  onSubmit: (e: SyntheticEvent) => void
+  query: SearchQuery
+  setAutocompleteEnabled: React.Dispatch<React.SetStateAction<boolean>>
+}) => {
+  const onSelectVehicleOption = (vehicle: Vehicle | Ghost) => {
+    dispatch(
+      setSelectedEntity({
+        type: SelectedEntityType.Vehicle,
+        vehicleId: vehicle.id,
+      })
+    )
+  }
+
+  const onSelectedLocationId = (id: string) => {
+    dispatch(
+      setSelectedEntity({
+        type: SelectedEntityType.LocationByPlaceId,
+        placeId: id,
+      })
+    )
+  }
+
+  const onSelectedLocationText = (text: string) => {
+    dispatch(setSearchText(text))
+    dispatch(submitSearch())
+  }
+
+  return (
+    <SocketContext.Provider value={useSocket()}>
```

Comment:
question: what's going on here? initially, this seems unnecessary?

Reply:
omg I forgot about this, my bad, I didn't see that this existed originally in the place it was extracted from. When I built this originally, I hated having to add this hack to make the sockets work 😭 Which I guess is why I questioned why it was here 😅